src/sensor.rs

```diff
@@ -16,10 +16,23 @@ const CO2_SENSOR_PRODUCT: u16 = 0xa052;
 const METER_CO2: u8 = 0x50;
 const METER_TEMP: u8 = 0x42;
 
+/// iSerial as reported by the usb device.
+///
+/// It doesn't seem to be a per-device serial number, but rather an extra model specifier. We use
+/// this to differentiate whether we need to generate and upload a key or use unencrypted
+/// communication.
+#[derive(Debug, PartialEq, Eq)]
+enum Version {
+    /// Version "1.00", uses encryption
+    V1_00,
```

Comment:
just noticed i mixed up `bcdDevice` and `iSerial` and their versions.. need to fix that.